server/graphql/invoice_line/src/mutations/outbound_shipment_line/save_item_lines.rs

```diff
@@ -0,0 +1,108 @@
+use async_graphql::*;
+
+use graphql_core::standard_graphql_error::{validate_auth, StandardGraphqlError};
+use graphql_core::ContextExt;
+use graphql_types::types::InvoiceNode;
+
+use repository::Invoice;
+use service::auth::{Resource, ResourceAccessRequest};
+use service::invoice_line::save_outbound_shipment_item_lines::{
+    SaveOutboundShipmentItemLines, SaveOutboundShipmentLine, SaveOutboundShipmentLinesError,
+};
+
+#[derive(InputObject)]
+pub struct SaveOutboundShipmentLinesInput {
+    pub invoice_id: String,
+    pub item_id: String,
+    pub lines: Vec<OutboundShipmentLineInput>,
+    pub placeholder_quantity: Option<f64>,
+}
+
+#[derive(InputObject)]
+pub struct OutboundShipmentLineInput {
+    pub id: String,
+    pub number_of_packs: f64,
+    pub stock_line_id: String,
+}
+
+pub fn save_outbound_shipment_item_lines(
+    ctx: &Context<'_>,
+    store_id: &str,
+    input: SaveOutboundShipmentLinesInput,
+) -> Result<InvoiceNode> {
+    let user = validate_auth(
+        ctx,
+        &ResourceAccessRequest {
+            resource: Resource::MutateOutboundShipment,
+            store_id: Some(store_id.to_string()),
+        },
+    )?;
+
+    let service_provider = ctx.service_provider();
+    let service_context = service_provider.context(store_id.to_string(), user.user_id)?;
+
+    map_response(
+        service_provider
+            .invoice_line_service
+            .save_outbound_shipment_item_lines(&service_context, input.to_domain()),
+    )
+}
+
+pub fn map_response(from: Result<Invoice, SaveOutboundShipmentLinesError>) -> Result<InvoiceNode> {
+    let result = match from {
+        Ok(invoice) => InvoiceNode::from_domain(invoice),
+        Err(error) => map_error(error)?,
+    };
+
+    Ok(result)
+}
+
+impl SaveOutboundShipmentLinesInput {
+    pub fn to_domain(self) -> SaveOutboundShipmentItemLines {
+        let SaveOutboundShipmentLinesInput {
+            invoice_id,
+            item_id,
+            lines,
+            placeholder_quantity,
+        } = self;
+
+        SaveOutboundShipmentItemLines {
+            invoice_id,
+            item_id,
+            placeholder_quantity,
+            lines: lines
+                .into_iter()
+                .map(|line| SaveOutboundShipmentLine {
+                    id: line.id,
+                    number_of_packs: line.number_of_packs,
+                    stock_line_id: line.stock_line_id,
+                })
+                .collect(),
+        }
+    }
+}
+
+fn map_error(error: SaveOutboundShipmentLinesError) -> Result<InvoiceNode> {
+    use SaveOutboundShipmentLinesError::*;
+    let formatted_error = format!("{:#?}", error);
+
+    log::error!("Error: {}", formatted_error);
+
+    // Future TODO: Implement structured errors where needed
+    // (Would only occur if 2 people editing at same time)
```

Comment:
Were returning all the structured errors in the old mutation - but weren't handling them on frontend? So I figured this was simpler, until there are actually errors we want to translate!